hedera-node/docs/design/services/smart-contract-service/contract-accounts-nonces.md

```diff
@@ -0,0 +1,64 @@
+| EOA transaction (`ContractCall` or `ContractCreate`)                                    | -                                                                                                          | -                                                                                                                                                                                | -                                                                                                                                                                                |
+| Contract transaction resulting in `CREATE/CREATE2` (`EthereumTransaction`)              | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+| Contract transaction resulting in `CREATE/CREATE2` (`ContractCall` or `ContractCreate`) | -                                                                                                          | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+
+
+
+### Contract Nonce Externalization
+
+- We can keep a `ContractId -> nonce` map inside `TxnAwareRecordsHistorian`, it would be updated on each call to `AbstractRecordingCreateOperation` (possibly using its `createOperationExternalizer` field) even if the contract creation operation does not succeed
+- When the ledgers `commit()` are executed, and we go to `TxnAwareRecordsHistorian.saveExpirableTransactionRecords` method we can save the constructed map to the top-level `RecordStreamObject`
+- This would be done via setting it inside it's `ExpirableTxnRecord` in the corresponding `EvmFnResult` field
+    - `contractCreateResult` if the top level transaction was `ContractCreate`
+    - `contractCallResult` if the top level transaction was `ContractCall`
+
+### Merging of contract into hollow account
+
+When contract is merged into an existing hollow account we are correctly setting its nonce value to 1 in `HederaCreateOperationExternalizer.finalizeHollowAccountIntoContract` as specified in [EIP-161](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-161.md).
+
+### New Classes
+
+No new classes expected to be created as part of this design.
+
+## Acceptance Tests
+
+### Positive Tests
```

Comment:
As a test case for multi creates e.g. Contract A with nonce 2, calls contract B with nonce 1 that creates Contract C and Contract D
The ContractFunctionResult should have
- B -> 3
- C -> 1
- D -> 1
And verify that A is unchanged still at 2

Reply:
Added.